src/camunda_worker/tasks/tasks/celery.py

```diff
@@ -0,0 +1,59 @@
+from django.conf import settings
+
+from celery.utils.log import get_task_logger
+
+from camunda_worker.external_tasks.camunda import fetch_and_lock
+from camunda_worker.external_tasks.constants import Statuses
+from camunda_worker.external_tasks.models import FetchedTask
+
+from ...celery import app
+from ..api import complete, execute
+
+logger = get_task_logger(__name__)
+
+__all__ = ("task_fetch_and_lock", "task_execute_and_complete")
+
+
+@app.task()
+def task_fetch_and_lock():
+    worker_id, num_tasks, tasks = fetch_and_lock(settings.MAX_TASKS)
+
+    logger.info("fetched %r tasks with %r", num_tasks, worker_id)
+
+    for task in tasks:
+        task_execute_and_complete(task.id)
```

Comment:
```suggestion
        task_execute_and_complete.delay(task.id)
```

I think the fetching and locking should be decoupled from the actual execution - now the execution is done synchronously as part of the fetch and lock task.